src/cpuid/src/transformer/intel.rs

```diff
@@ -118,6 +118,28 @@ fn update_extended_topology_entry(
 pub struct IntelCpuidTransformer {}
 
 impl CpuidTransformer for IntelCpuidTransformer {
+    fn process_cpuid(&self, cpuid: &mut CpuId, vm_spec: &VmSpec) -> Result<(), Error> {
+        // The following commit changed the behavior of KVM_GET_SUPPORTED_CPUID to no longer
+        // include leaf 0xb / sub-leaf 1.
+        // https://lore.kernel.org/all/20221027092036.2698180-1-pbonzini@redhat.com/
+        if !cpuid
+            .as_slice()
+            .iter()
+            .any(|entry| entry.function == 0xb && entry.index == 1)
+        {
+            cpuid
+                .push(kvm_cpuid_entry2 {
+                    function: leaf_0xb::LEAF_NUM,
+                    index: 1,
+                    flags: kvm_bindings::KVM_CPUID_FLAG_SIGNIFCANT_INDEX,
+                    ..kvm_cpuid_entry2::default()
+                })
+                .map_err(Error::Fam)?;
```

Comment:
I believe it is important this is inserted with zero values? In this case where we care about the value, we should specifically use 0 rather than the default (which could change in the future).
```suggestion
            cpuid
                .push(kvm_cpuid_entry2 {
                    function: leaf_0xb::LEAF_NUM,
                    index: 1,
                    flags: kvm_bindings::KVM_CPUID_FLAG_SIGNIFCANT_INDEX,
                    eax: 0,
                    ebx: 0,
                    ecx: 0,
                    edx: 0,
                    padding: Default::default()
                })
                .map_err(Error::Fam)?;
```

Reply:
Since EAX through EDX values are overwritten in `update_extended_topology_entry()`, any values are acceptible and don't matter here, I think. What do you think?

Reply:
You are right. It may be worth adding a comment on this to be clear, like:

> The registers within sub-leaves of `0xB` are zeroed by `update_extended_topology_entry()` and then modified. So the value they are set to at this point doesn't matter.

Reply:
Looks better! Thanks!